### ISSUE TITLE
Telemetry integration tests fail in istiodremote topology

### ISSUE THREAD

**Issue by frankbu** [istio/istio#35192]
### Bug Description

Many of the telemetry tests are failing when trying to run them using the istiodremote topology. Most failures include an error message something like this:
```
logs: did not get expected log entry:
got []
want http_request:{request_method:"GET"  request_url:"http://srv.istio-echo-2-22342.svc.cluster.local/foo"  status:200 ...
```
The consistent `got []` seems to imply that the log isn't set up or enabled properly.

The following command can be used to run the tests locally:
```
sudo make shell

export HUB=gcr.io/istio-testing
export TAG=1.12-alpha.ce6981cec04932b4f4dc38499efcb708b22eb013
rm -rf artifacts

ARTIFACTS=$PWD/artifacts ./prow/integ-suite-kind.sh --topology MULTICLUSTER --skip-build --skip-cleanup --topology-config prow/config/topology/external-istiod.json
go test -p 1 -vet=off -v -count=1 -tags=integ ./tests/integration/telemetry/... -timeout 30m --istio.test.ci --istio.test.pullpolicy=IfNotPresent --istio.test.kube.topology=/work/localtest.external-istiod.json --istio.test.skipVM > /tmp/log_telemetry.txt 2>&1
```

### Version

```prose
latest
```

**Comment by frankbu** [comment]
Prow job: https://prow.istio.io/?job=integ-telemetry-istiodremote-tests_istio_postsubmit

**Comment by frankbu** [comment]
Telemetry was not enabled, which explains many of the failures. [This PR enables telemetry](https://github.com/istio/istio/pull/35280) after which the tests get much further, but there are [still 6/40 failing tests](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/35280/integ-telemetry-istiodremote-tests_istio/1440001737535524864).

**Comment by frankbu** [comment]
With the fixes in #35316, we now only have one remaining failure: https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/35316/integ-telemetry-istiodremote-tests_istio/1440751041216778240

This is a setup error for the stackdriver suite, which looks like it might be the same problem as is causing suite setup errors in the security tests #35204

**Comment by johnma14** [comment]
Thanks @frankbu, looking into these failing tests.

**Comment by frankbu** [comment]
Fixed by #35422 

https://prow.istio.io/view/gs/istio-prow/logs/integ-telemetry-istiodremote-tests_istio_postsubmit/1443587158383267840

**Comment by johnma14** [comment]
Awesome. Thanks @frankbu
